arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/EagerEffect.kt

```diff
@@ -115,6 +115,11 @@ public interface EagerEffect<out R, out A> {
       toEither().fold({ r -> f(r).bind() }, ::identity)
     }
 
+  public fun <R2> mapLeft(f: (R) -> R2): EagerEffect<R2, A> =
+    handleErrorWith {
+      eagerEffect { shift(f(it)) }
+    }
```

Comment:
a more efficient implementation is
```suggestion
  public fun <R2> mapLeft(f: (R) -> R2): EagerEffect<R2, A> =
    eagerEffect {
      toEither().fold({ shift(f(it)) }, ::identity)
    }
```

Reply:
I meant efficient in terms of the encoding, and specifically for mapLeft not creating too many continuations. 

> And speaking about efficiency, is ::identity less or more efficient than a simple { it } or why is it preferred?
same answer as above and ::identity is rather a convention both suffice 

> Also why does EagerEffect need the toEither() while Effect doesn't?
That has to do with `shift` being suspend and fold in EagerEffect is not inline

regarding https://github.com/arrow-kt/arrow/pull/2765 can you reproduce what you mean/or show what crashes in an example repo?

Reply:
ok, I will change, although I don't see how it could create more continuations. It could at most add (if not optimized by the compiler) an extra function call due to implementing it in terms of the other operator, right? Also `handleErrorWith` is not even `suspend` so I don't see how it could involve any differences in terms of continuations.

I am not sure about the `::identity` part if you mean I should use it cause it is an Arrow convention (I guess it is not a Kotlin one) or if  `{ it }` is also fine. I will go with `{ it }` on first update, change to `::identity` afterwards if you meant that was better / needed too.

And about #2765 I just mentioned it here cause it makes me feel uneasy about the `::identity`. But yes, as I said I'm trying to come up with some example for the #2765 follow up (#2765 did improve things, but I still have some other crashes which look very similar in the stacktraces). It is just not trivial to figure out the exact minimal triggers since I get it only on some rather complex part of my code mixed with things that are not really good to have on an issue report so I'm trying to get rid of that "noise" while keeping the issue happening and I have not been yet able to...